ci: make cache works by removing comments

ylz-at opened the pull request:
The path is a multiline string, which includes comments in its value. This
means that the caching was not working at all, because it would include
the comment in the path. Here I'm moving the comments, which fixes the
caching so it works.

<!--
Checklist:

- API changes have been discussed
- code is goformated correctly (goimports)
- packages with generated code have had code generation run
- tests pass locally
- linked to relevant issues

Please make sure your commit message summary line and pull request
title match the Go convention; a one-line summary of the change,
prefixed by the primary affected package that should complete the
sentence, "This change modifies Gonum to _____."
-->

kortschak commented:
Thank you. Would you please also send another PR adding yourself to AUTHORS and CONTRIBUTORS with the commit message "A+C: add Jinesi Yelizati".

kortschak reviewed (approved):
LGTM but waiting for A+C PR.

ylz-at commented:
Hi, I updated actions/cache to v2 as v1 doesn't work with multiline string.
And I moved `Checkout code` before `Cache-Go` as `Cache-Go` depends on `go.sum`.

[Have tested on my own repo](https://github.com/ylz-at/gonum/runs/3713059510?check_suite_focus=true), Cache-Go can work this time.

kortschak reviewed (approved):
Thanks
